x-pack/legacy/plugins/apm/public/projections/metrics.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Setup } from '../../server/lib/helpers/setup_request';
+import {
+  SERVICE_NAME,
+  PROCESSOR_EVENT
+} from '../../common/elasticsearch_fieldnames';
+import { rangeFilter } from '../../server/lib/helpers/range_filter';
+
+export function getMetricsProjection({
+  setup,
+  serviceName
+}: {
+  setup: Setup;
+  serviceName: string;
+}) {
+  const { start, end, uiFiltersES, config } = setup;
+
+  return {
+    index: config.get<string>('apm_oss.metricsIndices'),
```

Comment:
Just a thought: should we make this a constant like:
```ts
index: config.get<string>(APM_OSS_METRICS_INDICES)
```

or perhaps:
```ts
index: getMetricsIndicesName(config)
```

Reply:
(not really related to this PR)

Reply:
Aye, I think constants would be nice. I'll leave it out of this PR though.